lib/async_hooks.js

```diff
@@ -268,15 +269,27 @@ const storageHook = createHook({
     const currentResource = executionAsyncResource();
     // Value of currentResource is always a non null object
     for (let i = 0; i < storageList.length; ++i) {
-      storageList[i]._propagate(resource, currentResource);
+      storageList[i]._propagate(resource, currentResource, type);
     }
   }
 });
 
 class AsyncLocalStorage {
-  constructor() {
+  constructor(options = {}) {
```

Comment:
`kEmptyObject` from `'internal/util'` is preferred to avoid prototype pollutions.

```suggestion
  constructor(options = kEmptyObject) {
```